### PR TITLE
Add top level common parameters

### PR DESCRIPTION
**Pull request by dschwen**
Add the ability to specify common parameters for subblocks in the TensorMechanics Master action.

Refs #7987

### CODE REVIEW

File: modules/tensor_mechanics/src/base/TensorMechanicsApp.C [hunk 1]
```diff
@@ -380,6 +381,7 @@ TensorMechanicsApp::associateSyntax(Syntax & syntax, ActionFactory & action_fact
   syntax.registerActionSyntax("PressureAction", "BCs/Pressure/*");
 
   syntax.registerActionSyntax("GeneralizedPlaneStrainAction", "Modules/TensorMechanics/GeneralizedPlaneStrain/*");
+  syntax.registerActionSyntax("CommonTensorMechanicsAction", "Modules/TensorMechanics/Master");
```
Comment (bwspenc) [inline modules/tensor_mechanics/src/base/TensorMechanicsApp.C]: Couldn't you just register `TensorMechanicsActionBase` here and get rid of `CommonTensorMechanicsAction`?

**Review by bwspenc — COMMENTED** [review]
I really like what this does to the input syntax, and I'm generally happy with this.  I'm still just struggling a little bit to see the purpose of having both `CommonTensorMechanicsAction` and `TensorMechanicsActionBase`.  If we could get by with just having one of them, I think it would make the code easier to understand.

**Comment by dschwen** [comment]
I don't think we can do this. And if we figure out how to do it it will
make the code even more complicated. Let's talk on Monday.

File: modules/tensor_mechanics/src/base/TensorMechanicsApp.C (hunk 1 above)
Comment (dschwen) [inline modules/tensor_mechanics/src/base/TensorMechanicsApp.C]: I don't know if the action fetching works if I do that! The TM action will also dynamic cast to tmbase.

Comment (permcody) [inline modules/tensor_mechanics/src/base/TensorMechanicsApp.C]: Yeah that doesn't work. The Action system needs to know what kind of instance to build when encountering a piece of syntax. Having the Base registered means the Action system would try to build the base type directly.

**Review by permcody — CHANGES_REQUESTED** [review]
This PR looks OK but there's no new documentation here for this important capability. I'd suggest putting in detailed comments in the test for a start but also documenting how this works in the higher level documentation too.

**Review by bwspenc — APPROVED** [review]

**Comment by bwspenc** [comment]
@permcody he added some documentation. I'm good with the code as it is now.  If you're good with it, can you accept those changes?

**Review by permcody — APPROVED** [review]

**Pull request merged**